log_shipping/clent_log_shipping_s3_bucket_create.go

```diff
@@ -0,0 +1,81 @@
+package log_shipping
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	logzio_client "github.com/logzio/logzio_terraform_client"
+)
+
+const createS3BucketServiceMethod string = http.MethodPost
+const createS3BucketMethodSuccess int = http.StatusOK
+
+// type FieldError struct {
+// 	Field   string
+// 	Message string
+// }
+
+// func (e FieldError) Error() string {
+// 	return fmt.Sprintf("%v: %v", e.Field, e.Message)
+// }
+
+func validateCreateS3BucketRequest(bucket S3Bucket) error {
+
+	if len(bucket.AccessKey) == 0 {
+		return fmt.Errorf("accessKey must be set")
+	}
+
+	// TODO: complete validation
```

Comment:
What else needs to be validated?

Reply:
I was wondering if this should duplicate validation done by the API. It seems like some parts of the codebase attempt to duplicate API-level validation, so that Terraform fails during `plan` rather than during `apply`. However I'll leave that out for now as it adds complexity for uncertain benefit.